ccl/multiregionccl: TestAddRegionSucceedWithConcurrentBackupOps failed [test timeout]

ccl/multiregionccl.TestAddRegionSucceedWithConcurrentBackupOps [failed](https://tanzanite.cluster.engflow.com/invocations/default/f9012730-be92-45bb-a41b-a0e08397516b?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [9354770c7c6eb5a89437068d8c6a4accf8031b67](https://github.com/cockroachdb/cockroach/commits/9354770c7c6eb5a89437068d8c6a4accf8031b67):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestAddRegionSucceedWithConcurrentBackupOps (39m17s)
	TestAddRegionSucceedWithConcurrentBackupOps/backup-database (39m17s)

```
Stack: 

```
goroutine 119876029 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAddRegionSucceedWithConcurrentBackupOps
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAddRegionSucceedWithConcurrentBackupOps1589788319
    test_log_scope.go:76: use -show-logs to present logs inline
=== RUN   TestAddRegionSucceedWithConcurrentBackupOps/backup-database
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAddRegionSucceedWithConcurrentBackupOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45637

--- a/pkg/ccl/multiregionccl/region_test.go
+++ b/pkg/ccl/multiregionccl/region_test.go
@@ -924,6 +924,7 @@ func testRegionAddDropWithConcurrentBackupOps(
 	},
 ) {
 	skip.UnderRace(t, "times out under race")
+	skip.UnderDeadlock(t)
 
 	testCases := []struct {
 		name      string

